src/PanelTraits/FakeFields.php

```diff
@@ -66,4 +69,16 @@ public function compactFakeFields($request, $form = 'create', $id = false)
         // since no modifications or additions have been made to $request
         return $request;
     }
+
+    /**
+     * @return array
+     */
+    protected function getModelCasts()
+    {
+        $modelReflection = new \ReflectionClass($this->model);
+        $castsAttribute = $modelReflection->getProperty('casts');
+        $castsAttribute->setAccessible(true);
```

Comment:
Should a function conspicuously named **get** be changing the accessibility of an attribute?

Better yet, why?

In fact, if this really does only get the casts it shouldn't be making them accessible - I'd suggest refactoring that out to another method or calling it straight after the `getModelCasts()` above unless there's a compelling reason to keep it here [in which case it probably needs a comment to stop others wondering why `get` functions are running about changing accessible casts].

Reply:
Why is that `setAccessible(true)` there - I'd be inclined to comment it.